dm/dm/master/server.go

```diff
@@ -1070,6 +1068,47 @@ func (s *Server) getStatusFromWorkers(ctx context.Context, sources []string, tas
 	}
 	wg.Wait()
 	s.fillUnsyncedStatus(workerResps)
+
+	getSubTaskStatusCountInResp := func(taskName string) int {
+		cnt := 0
+		for _, resp := range workerResps {
+			for _, status := range resp.SubTaskStatus {
+				if status.Name == taskName {
+					cnt += 1
+				}
+			}
+		}
+		return cnt
+	}
+	appendFakeResp := func(taskName, sourceName, msg string, cnt int) {
+		for i := 0; i < cnt; i++ {
+			setWorkerResp(&pb.QueryStatusResponse{
+				Result: false, Msg: msg, SubTaskStatus: []*pb.SubTaskStatus{{Name: taskName}},
+				SourceStatus: &pb.SourceStatus{Source: sourceName, Worker: "unknown-offline-worker"},
+			})
+		}
+	}
+
+	// when taskName is empty we need list all task even the worker that handle this task is not running.
+	if taskName == "" {
+		for taskName, sourceM := range s.scheduler.GetSubTaskCfgs() {
+			msg := fmt.Sprintf("can't find task: %s from dm-worker, please use dmctl list-member to check if worker is offline.", taskName)
```

Comment:
```suggestion
			msg := fmt.Sprintf("can't find task %s from dm-worker, please use dmctl list-member to check if worker is offline.", taskName)
```

Reply:
fixed in https://github.com/pingcap/ticdc/pull/3358/commits/f6079ca5bb882a49f8e06cfefde2a7620ab1b1e0

> why do we choose this way instead of asking all sources
becasue if some worker is offline, we can't get the worker_name from `s.scheduler`

Reply:
> why do we choose this way instead of asking all sources
becasue if some worker is offline, we can't get the worker_name from s.scheduler

s.scheduler has all registered workers. I think scheduler has the worker_name 🤔

Reply:
yes but we lost the source to worker bound relationships, so it's hard to find the right worker that used to handle this task